ssl/statem/extensions_srvr.c

```diff
@@ -196,7 +196,8 @@ int tls_parse_ctos_maxfragmentlen(SSL_CONNECTION *s, PACKET *pkt,
      * including session resumptions.
      * We should receive the same code as in resumed session !
      */
-    if (s->hit && s->session->ext.max_fragment_len_mode != value) {
+    if (s->session->ext.max_fragment_len_mode != TLSEXT_MFL_UNSPECIFIED
```

Comment:
Speaking very naively here, but, gien that max_fragment_len_mode gets set based on the parsing of extensions, doesn't control flow through this function get altered based on the order in which those extensions are processed? Is that the intent, or is it irrelevant?

Reply:
So long as any session ticket extension is processed first, the rest don't matter. In any case, I do think that once we stop being needlessly (erroneously) pedantic, the logic changes a lot, because we either use the new value if not resuming, or keep the old value otherwise, and there's nothing to ever reject (based on this extension and values or not from a previous session).